device-to-group-command.py

```diff
@@ -0,0 +1,390 @@
+
+def get_whitelist_api_url(device_id):
+    return configuration.host + '/enterprise/' + enterprise_id + '/device/' + device_id + '/app/'
+
+def is_package_whitelisted(device_id, package_name):
+    url = get_whitelist_api_url(device_id)
+    querystring = {"package_name": package_name, "whitelisted":"false"}
+    payload = ""
+    headers = {
+        'Authorization': "Bearer " + configuration.api_key['Authorization'],
+        'cache-control': "no-cache",
+        }
+
+    response = requests.request("GET", url, data=payload, headers=headers, params=querystring)    
+    print("Whitelist query match: " + str(response.json().get('count')))
+    return False if response.json().get('count') == 1 else True
+
+
+def whitelist_package_in_group_devices(device_list, package_name):
+    print("Currently active devices in group: ")
+    print([device.device_name for device in device_list])
+
+    for device in device_list:
+        if not is_package_whitelisted(device.id, package_name):
+            try:
+                run_whitelist_command(device, package_name)
+            except Exception as e:
+                print('Could not fire command on ' + device.device_name + ": " + str(e))
+            time.sleep(1)
+        else:
+            print('Warning: Package already whitelisted or not present on device: ' + device.device_name)
+
+
+# ====== UNINSTALL FROM GROUP ==========            
+
+
+def run_uninstall_command(device, package_name, app_id):
+    api_instance = esperclient.CommandsApi(esperclient.ApiClient(configuration))
+    command_args = esperclient.CommandArgs(package_name=package_name, app_version = app_id)
+    command = esperclient.CommandRequest(command='UNINSTALL', command_args=command_args)
+    try:
+        api_response = api_instance.run_command(enterprise_id, device.id, command)
+        if api_response.id is not None:
+            print('Uninstall command fired successfully on ' +  device.device_name + ' for ' + package_name)
+    except ApiException as e:
+        print("Exception when calling CommandsApi->run_command: %s\n" % e)
+
+
+def uninstall_package_in_group_devices(device_list, package_name):
+
+    print("Currently active devices in group: ")
+    print([device.device_name for device in device_list])
+
+    for device in device_list:
+        app_id = get_package_id(device.id, package_name)
+        if app_id:
+            try:
+                run_uninstall_command(device, package_name, app_id)
+            except Exception as e:
+                print('Could not fire command on ' + device.device_name + ": " + str(e))
+            time.sleep(1)
+        else:
+            print('Warning: Package already uninstalled or not present on device: ' + device.device_name)
+
+# This function returns the package app id if installed else return null
+def get_package_id(device_id, package_name):
+    querystring = {"package_name": package_name, "whitelisted":"false"}
+    payload = ""
+    headers = {
```

Comment:
Remove headers, payload and commented code (below). Its not being used in this code.